src/thriveopedia/Thriveopedia.cs

```diff
@@ -618,24 +619,47 @@ private void OnForwardPressed()
 
     private void OnSearchUpdated(string newText)
     {
-        pageTree.Clear();
-        pageTree.CreateItem();
+        stageDropdown.Visible = false;
 
-        // Need to copy to new list as loop modifies the list
-        var pageList = allPages.Keys.ToList();
-        foreach (var page in pageList)
+        foreach (var page in allPages)
         {
-            var children = GetAllChildren(page);
+            var visible = page.Key.TranslatedPageName.ToLower(CultureInfo.CurrentCulture)
+                .Contains(newText.ToLower(CultureInfo.CurrentCulture));
```

Comment:
One more comment on this line. Why isn't the `newText.ToLower...` call outside the loop? Surely that cannot change each loop iteration and this way we save on creating new strings from the new text ToLower call once per loop iteration.